src/legacy/ui/public/vislib/visualizations/point_series/column_chart.js

```diff
@@ -236,13 +304,44 @@ export class ColumnChart extends PointSeries {
       return Math.abs(yScale(baseValue) - yScale(d.y));
     }
 
+    function formatValue(d) {
+      return chartData.yAxisFormatter(d.y);
+    }
+
     // update
     bars
       .attr('x', isHorizontal ? x : y)
       .attr('width', isHorizontal ? widthFunc : heightFunc)
       .attr('y', isHorizontal ? y : x)
       .attr('height', isHorizontal ? heightFunc : widthFunc);
 
+    const layer = d3.select(bars[0].parentNode);
+    const barLabels = layer.selectAll('text').data(chartData.values.filter(function (d) {
+      return !_.isNull(d.y);
+    }));
+
+    barLabels
+      .exit()
+      .remove();
+
+    if (isLabels) {
+      const labelColor = this.handler.data.getColorFunc()(chartData.label);
+
+      barLabels
+        .enter()
+        .append('text')
+        .text(formatValue)
+        .attr('class', 'bar-label--normal')
```

Comment:
Can you use a proper BEM naming scheme for all the new classes with the appropriate `vis` prefix? Something like `.visColumnChart__barLabel--normal`. This will ensure scope and understanding of where the styles are being applied.